src/Core/Settings.h

```diff
@@ -277,6 +277,8 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(Milliseconds, stream_flush_interval_ms, 7500, "Timeout for flushing data from streaming storages.", 0) \
     M(Milliseconds, stream_poll_timeout_ms, 500, "Timeout for polling data from/to streaming storages.", 0) \
     \
+    M(Bool, force_select_final, false, "Query with the FINAL modifier by default", 0) \
```

Comment:
As we have the `offset` and `limit` settings for "out of band" query modification, we can name the new setting simply `final`.

The documentation has to be improved.